examples/freesolv/freesolv_testsystem.py

```diff
@@ -0,0 +1,172 @@
+import numpy as np
+from perses.rjmc import geometry
+from openmmtools import states, mcmc
+from simtk import openmm, unit
+from simtk.openmm import app
+from openmmtools import cache
+from perses.storage import NetCDFStorageView, NetCDFStorage
+from perses.rjmc.topology_proposal import SmallMoleculeSetProposalEngine
+from typing import List, Dict
+
+cache.global_context_cache.platform = openmm.Platform.getPlatformByName("CUDA")
+cache.global_context_cache.capacity = 30
+
+
+class HydrationPersesRun(object):
+    """
+    This class encapsulates the necessary objects to run RJ calculations for relative hydration free energies
+    """
+
+    def __init__(self, molecules: List[str], output_filename: str, ncmc_switching_times: Dict[str, int], equilibrium_steps: Dict[str, int], timestep: unit.Quantity, initial_molecule: str=None, geometry_options: Dict=None):
+        self._molecules = [SmallMoleculeSetProposalEngine.canonicalize_smiles(molecule) for molecule in molecules]
+        environments = ['explicit', 'vacuum']
+        temperature = 300 * unit.kelvin
+        pressure = 1.0 * unit.atmospheres
+        constraints = app.HBonds
+        self._storage = NetCDFStorage(output_filename)
+        self._ncmc_switching_times = ncmc_switching_times
+        self._n_equilibrium_steps = equilibrium_steps
+        self._geometry_options = geometry_options
+
+        # Create a system generator for our desired forcefields.
+        from perses.rjmc.topology_proposal import SystemGenerator
+        system_generators = dict()
+        from pkg_resources import resource_filename
+        gaff_xml_filename = resource_filename('perses', 'data/gaff.xml')
+        barostat = openmm.MonteCarloBarostat(pressure, temperature)
+        system_generators['explicit'] = SystemGenerator([gaff_xml_filename, 'tip3p.xml'],
+                                                        forcefield_kwargs={'nonbondedMethod': app.PME,
```

Comment:
You might also have to set the PME tolerance to 1e-5 instead of 5e-5! That's `ewaldErrorTolerance` in kwargs.

Reply:
Also, I'm not sure if we've been strict about keeping the integrator constraint tolerance to 1e-8 to ensure iteration to convergence, so let's double-check that.